src/containers/MessengerContainer.tsx

```diff
@@ -80,83 +66,140 @@ export const MessengerContainer: FC = () => {
     syncToWidth(); // run once on mount
     window.addEventListener("resize", syncToWidth);
     return () => window.removeEventListener("resize", syncToWidth);
-  }, [
-    messageStore.openedRecipient,
-    isMobile,
-    isBroadcastMode,
-    selectedChannelName,
-  ]);
+  }, [contactId, channelId, isMobile, isCurrentlyInBroadcastMode]);
+
+  // store cleanup functions in refs to avoid re-renders
+  const cleanupFnsRef = useRef({
+    messageStore,
+    walletStore,
+    uiStore,
+    closeAllModals,
+    setAttachment,
+  });
 
-  // Clean up useEffect
+  // update refs when dependencies change
+  useEffect(() => {
+    cleanupFnsRef.current = {
+      messageStore,
+      walletStore,
+      uiStore,
+      closeAllModals,
+      setAttachment,
+    };
+  });
+
+  // clean up useeffect
   useEffect(() => {
     return () => {
-      messageStore.stop();
+      const fns = cleanupFnsRef.current;
+      fns.messageStore.stop();
 
-      // Called when MessagingContainer unmounts (user leaves route), so we can reset all the states
-      walletStore.lock();
-      uiStore.setSettingsOpen(false);
-      closeAllModals();
+      // called when messagingcontainer unmounts (user leaves route), so we can reset all the states
+      fns.walletStore.lock();
+      fns.uiStore.setSettingsOpen(false);
+      fns.closeAllModals();
 
-      messageStore.setOpenedRecipient(null);
+      fns.messageStore.setOpenedRecipient(null);
 
-      // Clear broadcast store state when leaving messaging
+      // clear broadcast store state when leaving messaging
       useBroadcastStore.getState().reset();
 
       // clear any attachments
-      setAttachment(null);
+      fns.setAttachment(null);
     };
   }, []);
 
-  useEffect(() => {
-    if (
-      isWalletReady &&
-      (!messageStore.isLoaded || !walletStore.isAccountServiceRunning)
-    ) {
-      setLoadingMessage(loadingMessages[0].message);
-      const timeouts = loadingMessages
-        .slice(1)
-        .map(({ delay, message }) =>
-          setTimeout(() => setLoadingMessage(message), delay)
-        );
-      return () => {
-        timeouts.forEach(clearTimeout);
-      };
-    }
-  }, [
-    isWalletReady,
-    messageStore.isLoaded,
-    walletStore.isAccountServiceRunning,
-  ]);
-
-  // Effect to restore last opened conversation after messages are loaded (desktop only)
+  // effect to restore last opened conversation after messages are loaded (desktop only)
   useEffect(() => {
     if (
       !isMobile &&
       messageStore.isLoaded &&
       walletStore.isAccountServiceRunning &&
-      !messageStore.openedRecipient &&
-      messageStore.oneOnOneConversations.length > 0
+      !contactId && // no contact selected in url
+      messageStore.oneOnOneConversations.length > 0 &&
+      !isCurrentlyInBroadcastMode // only for direct messages
     ) {
```

Comment:
Will probably need a remake at some point, this effect is getting bigger and bigger and could become "uncontrollable" with "too-much" dependencies